Sources/Public/iOS/LottieAnimationViewBase.swift

```diff
@@ -39,7 +39,11 @@ open class LottieAnimationViewBase: UIView {
 
   var screenScale: CGFloat {
     #if os(iOS) || os(tvOS)
-    UIScreen.main.scale
+    if #available(iOS 13.0, tvOS 13.0, *) {
+      return UITraitCollection.current.displayScale
```

Comment:
From the documentation:

> A value of 1.0 indicates a non-Retina display and a value of 2.0 indicates a Retina display. The default display scale for a trait collection is 0.0 (indicating unspecified).

Can you manually check for the case where `displayScale` is zero, and make sure we always return at least `1`?

```suggestion
      return max(UITraitCollection.current.displayScale, 1)
```

Also, could you verify that `UITraitCollection.current.displayScale` returns `3.0` on a modern 3x Retina display? The documentation doesn't mention this explicitly, but may simply be outdated.

Reply:
I already have checked that returns 3.0 in 3x Retina display.

CC.https://www.ios-resolution.com

Reply:
I agree to outdated doc.
Will the document be updated, if visionOS beta is officially adopted?
I'm wondering like you!